src/nerve_tools/configure_nerve.py

```diff
@@ -204,43 +223,39 @@ def generate_configuration(services, heartbeat_path, hacheck_port, weight, zk_to
         for depth, loc in enumerate(available_locations)
     }
 
-    for (service_name, service_info) in services:
-        port = service_info.get('port')
-        if port is None:
-            continue
-
-        mode = service_info.get('mode', 'http')
-        healthcheck_timeout_s = service_info.get('healthcheck_timeout_s', 1.0)
-        healthcheck_port = service_info.get('healthcheck_port', port)
-
-        # hacheck will simply ignore the healthcheck_uri for TCP mode checks
-        healthcheck_uri = service_info.get('healthcheck_uri', '/status')
-        healthcheck_mode = service_info.get('healthcheck_mode', mode)
-        hacheck_uri = '/%s/%s/%s/%s' % (
-            healthcheck_mode, service_name, healthcheck_port, healthcheck_uri.lstrip('/'))
-        advertise = service_info.get('advertise', ['region'])
-        extra_advertise = service_info.get('extra_advertise', [])
-        extra_healthcheck_headers = service_info.get('extra_healthcheck_headers', {})
-
+    def update_subconfiguration_for_here(
+        service_name,
+        service_info,
+        service_weight,
+    ):
         nerve_config['services'].update(
             generate_subconfiguration(
                 service_name=service_name,
-                advertise=advertise,
-                extra_advertise=extra_advertise,
-                port=port,
+                service_info=service_info,
+                weight=service_weight,
                 ip_address=ip_address,
-                healthcheck_timeout_s=healthcheck_timeout_s,
-                hacheck_uri=hacheck_uri,
-                healthcheck_headers=extra_healthcheck_headers,
                 hacheck_port=hacheck_port,
-                weight=weight,
                 zk_topology_dir=zk_topology_dir,
                 zk_location_type=zk_location_type,
                 zk_cluster_type=zk_cluster_type,
                 location_depth_mapping=location_depth_mapping,
             )
         )
 
+    for (service_name, service_info) in classic_services:
+        update_subconfiguration_for_here(
+            service_name=service_name,
+            service_info=service_info,
+            service_weight=weight,
+        )
+
+    for (service_name, service_info) in paasta_services:
+        update_subconfiguration_for_here(
+            service_name=service_name,
+            service_info=service_info,
+            service_weight=1,
```

Comment:
How about instead of `1` you do `10` like what this script does when it can't get the number of cpus?

That way it will be ever so slightly closer to what we do with y-m to help in our transition, and if it is going to be an arbitrary number, 10 is just as good?